01_fib/shuutech/main_test.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNextVal(t *testing.T) {
+
+	var getNewPair = nextVal(0, 1, 2, 0)
+
+	var actual = getNewPair[0]
+	var expected = 1
+	if expected != actual {
+		t.Errorf("expected %v, got %v", actual, expected)
+		t.Fail()
```

Comment:
I believe `t.Fail` is not needed here, but please consult this with someone more experienced.